stdlib/test/builtin/test_string.mojo

```diff
@@ -98,6 +98,59 @@ fn test_equality_operators() raises:
     assert_not_equal(s0, "notabc")
 
 
+fn test_comparison_operators() raises:
+    var s0 = String("abc")
+    var s1 = String("def")
+    var s2 = String("abc")
+    var s3 = String("ABC")
+    var s4 = String("ab")
+    var s5 = String("abcd")
+    var empty = String("")  # Explicit empty string for comparison
+    var default_string = String()  # Default constructed, assumed empty
+
+    # Test less than and greater than
+    assert_true(s0 < s1)  # "abc" < "def"
+    assert_false(s1 < s0)  # "def" is not less than "abc"
```

Comment:
This PR is really cool! It might be possible that in the future, as devs add or modify tests, the comments get out of sync with the actual values of the strings. If you're worried about making long lines, what about this syntax?

```suggestion
    assert_false(str("def") < str("abc"))
```
Looking forward to your opinion on this :)

Reply:
I really like this suggestion, and I fully agree with what you're saying! My tests were pretty sloppily written so I'm gonna go fix this now :)

Reply:
I think tests always look a bit sloppy. But I do like a suggestion. Keep the tests as simple as possible.